_posts/2021-04-09-springboot-autoconfigurations.md

```diff
@@ -0,0 +1,436 @@
+# How Spring Boot's AutoConfigurations Internally Work
+
+In this article, we will discuss and understand how Spring Boot's AutoConfigurations internally works
+
+## 1. Introduction
+Before we deep dive into Spring Boot's AutoConfiguration, **what really is Spring Boot?**
+
+*As per Spring Boot documentation:*
+> It takes an opinionated view of the Spring platform and third-party libraries so you can get started with minimum fuss.
+
+## 2. What is Spring Boot Auto Configuration?
```

Comment:
Perhaps also mention in this section what you wanted to say in the previous section: that Auto-Configuration is one of the main features of Spring Boot to provide an opinionated view of third-party libraries